include/aspect/introspection.h

```diff
@@ -294,6 +294,28 @@ namespace aspect
       IndexSets index_sets;
 
       /**
+       * A struct that describes the available methods for
+       * compositional fields. This type is used to determine how
+       * to advect a compositional field.
+       */
+      struct FieldMethod
```

Comment:
I was playing with the thought of adding the option that the temperature field might be solved differently as well (as far as I know some codes in Europe do that), but currently it really only applies to compositional fields. The complication in finding a good name comes from the definition of "field", i.e. does it refer to one "composition" or does it refer to a "field method". Would you agree on CompositionalFieldMethod? I also thought about `AdvectionMethod`, but I really do not feel like any name is as descriptive as necessary.